hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -226,4 +254,37 @@ private void adjustFungibleTokenAllowance(final BalanceChange change, final Acco
         }
         accountsLedger.set(ownerID, FUNGIBLE_TOKEN_ALLOWANCES, fungibleAllowances);
     }
+
+    /**
+     * Checks if the alias is a known alias i.e, if the alias is already used in any cryptoTransfer
+     * transaction that has led to account creation
+     *
+     * @param change change that contains alias
+     */
+    private void checkIfExistingAlias(final BalanceChange change) {
+        if (change.hasNonEmptyAlias() && isKnownAlias(change.accountId(), currentView)) {
```

Comment:
This should already be a bit simpler with `AliasManager`; but adding `BalanceChange` methods like `@Nullable ByteString getNonEmptyAliasIfPresent()` and `void replaceNonEmptyAliasWith(EntityNum aliasNum)` would really clean this up.

Reply:
Fixed !